src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/RecordAudio/RecordAudio.hooks.ts

```diff
@@ -0,0 +1,96 @@
+import { useState, useCallback } from 'react';
+
+export interface recorderControls {
+  startRecording: () => void;
+  stopRecording: () => void;
+  togglePauseResume: () => void;
+  clearBlob: () => void;
+  recordingBlob: Blob[];
+  isRecording: boolean;
+  isPaused: boolean;
+  isStopped: boolean;
+  recordingTime: number;
+}
+
+export const useAudioRecorder: () => recorderControls = () => {
+  const [isRecording, setIsRecording] = useState(false);
+  const [isPaused, setIsPaused] = useState(false);
+  const [isStopped, setIsStopped] = useState(true);
+  const [recordingTime, setRecordingTime] = useState(0);
+  const [mediaRecorder, setMediaRecorder] = useState<MediaRecorder | null>();
+  const [timerInterval, setTimerInterval] = useState<NodeJS.Timer>();
+  const [recordingBlob, setRecordingBlob] = useState<Blob[]>([]);
+
+  const startTimer = () => {
+    const interval = setInterval(() => {
+      setRecordingTime((time) => time + 1);
+    }, 1000);
+    setTimerInterval(interval);
+  };
+
+  const stopTimer = () => {
+    timerInterval !== null && clearInterval(timerInterval);
+    setTimerInterval(undefined);
+  };
+
+  const startRecording = useCallback(() => {
+    if (timerInterval) return;
+
+    navigator.mediaDevices
+      .getUserMedia({ audio: true })
+      .then((stream) => {
+        setIsStopped(false);
+        setIsRecording(true);
+        const recorder: MediaRecorder = new MediaRecorder(stream);
+        setMediaRecorder(recorder);
+        recorder.start();
+        startTimer();
+
+        recorder.addEventListener('dataavailable', (event) => {
+          setRecordingBlob([...recordingBlob, event.data]);
+          // setRecordingBlob(event.data);
```

Comment:
Do we need this commented code?
If yes pls write according //TODO:

Reply:
removed